src/FlaUI.Adapter.White/Extensions/AdapterExtensions.ScrollableElement.cs

```diff
@@ -0,0 +1,185 @@
+            {
+                while (true)
+                {
+                    var beforePositionThumbPoint = positionThumb.BoundingRectangle.Center();
+                    lineDownButton.Invoke();
+                    if (!IsElementOffScreen(element))
+                    {
+                        return;
+                    }
+                    var afterPositionThumbPoint = positionThumb.BoundingRectangle.Center();
+                    if (beforePositionThumbPoint.X == afterPositionThumbPoint.X && beforePositionThumbPoint.Y == afterPositionThumbPoint.Y)
+                    {
+                        break;
+                    }
+                }
+            }           
+
+            // scroll up
+            var lineUpButton = element.GetScrollBarLineUpButton();
+            if (lineUpButton != null)
+            {
+                while (true)
+                {
+                    var beforePositionThumbPoint = positionThumb.BoundingRectangle.Center();
+                    lineUpButton.Invoke();
+                    if (!IsElementOffScreen(element))
+                    {
+                        return;
+                    }
+                    var afterPositionThumbPoint = positionThumb.BoundingRectangle.Center();
+                    if (beforePositionThumbPoint.X == afterPositionThumbPoint.X && beforePositionThumbPoint.Y == afterPositionThumbPoint.Y)
+                    {
+                        break;
+                    }
+                }
+            }           
+        }
+
+        private static Button GetScrollBarLineDownButton(this AutomationElement element)
+        {
+            var verticalScrollBar = element.GetVerticalScrollBar();
+            if(verticalScrollBar == null)
+            {
+                return null;
+            }
+            return verticalScrollBar.Get<Button>(SearchCriteria.ByControlType(ControlType.Button).AndByText("Line down"));
+        }
+
+        private static Button GetScrollBarLineUpButton(this AutomationElement element)
+        {
+            var verticalScrollBar = element.GetVerticalScrollBar();
+            if (verticalScrollBar == null)
+            {
+                return null;
+            }
+            return verticalScrollBar.Get<Button>(SearchCriteria.ByControlType(ControlType.Button).AndByText("Line up"));
+        }
+
+        private static AutomationElement GetScrollBarPositionThumb(this AutomationElement element)
+        {
+            var verticalScrollBar = element.GetVerticalScrollBar();
+            if (verticalScrollBar == null)
+            {
+                return null;
+            }           
+            return verticalScrollBar.Get(SearchCriteria.ByText("Position"));
+        }
+
+        private static AutomationElement GetVerticalScrollBar(this AutomationElement element)
+        {
+            return element.Get(SearchCriteria.ByControlType(ControlType.ScrollBar).AndByText("Vertical Scroll Bar"))
```

Comment:
This probably does not work in localized Windows. Maybe use the values from https://github.com/FlaUI/FlaUI/blob/master/src/FlaUI.Core/Tools/LocalizedStrings.cs ?

Reply:
fixed